common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java

```diff
@@ -215,91 +235,114 @@ private void tryAutoSearch(AbstractContainerScreen<?> abstractContainerScreen) {
         StyledText name = StyledText.fromComponent(abstractContainerScreen
                 .getMenu()
                 .getItems()
-                .get(currentInteractiveContainerType.getNextItemSlot())
+                .get(currentContainer.getNextItemSlot())
                 .getHoverName());
 
-        if (!name.matches(currentInteractiveContainerType.getNextItemPattern())) {
+        if (!name.matches(currentContainer.getNextItemPattern())) {
             autoSearching = false;
             return;
         }
 
         ContainerUtils.clickOnSlot(
-                currentInteractiveContainerType.getNextItemSlot(),
+                currentContainer.getNextItemSlot(),
                 abstractContainerScreen.getMenu().containerId,
                 GLFW.GLFW_MOUSE_BUTTON_LEFT,
                 abstractContainerScreen.getMenu().getItems());
     }
 
-    private InteractiveContainerType getCurrentInteractiveContainerType(Screen screen) {
-        InteractiveContainerType containerType = null;
-
-        for (InteractiveContainerType type : InteractiveContainerType.values()) {
-            if (type.isScreen(screen)) {
-                containerType = type;
+    private SearchableContainerProperty getCurrentSearchableContainer() {
+        // FIXME: Do something about this
```

Comment:
I guess you could put these in a `Map<Class<? extends SearchableContainerProperty>, Supplier<Boolean>>` but that wouldn't look much nicer, we would just move the code.

Reply:
I think this looks a lot better than all the if statements

Reply:
Indeed it looks better, all is well then.